public/app/core/components/NavBar/NavBarItem.tsx

```diff
@@ -50,7 +50,9 @@ const NavBarItem = ({
     }
 
     if (!target && url.startsWith('/')) {
-      locationService.push(url);
+      // when config.appSubUrl is configured the locationsService uses appSubUrl on all urls, so we need to strip it before push
+      const path = config.appSubUrl ? url.replace(config.appSubUrl, '') : url;
+      locationService.push(path);
```

Comment:
```suggestion
      locationService.push(locationUtil.stripBaseFromUrl(url));
```

Reply:
Commited suggestion, will clean up in a separate commit.